src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -358,9 +358,40 @@ public void EnqueueRange(IEnumerable<(TElement Element, TPriority Priority)> ite
                 throw new ArgumentNullException(nameof(items));
             }
 
+            int count = 0;
+            var collection = items as ICollection<(TElement Element, TPriority Priority)>;
+            if (collection is not null && (count = collection.Count) > _nodes.Length - _size)
+            {
+                Grow(_size + count);
+            }
+
             if (_size == 0)
             {
-                _nodes = EnumerableHelpers.ToArray(items, out _size);
```

Comment:
The issue I assume is that we're overwriting the existing array even if it would have been sufficient to store the data?

Reply:
Yeah exactly.